agent/system_metrics.go

```diff
@@ -0,0 +1,208 @@
+package agent
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"bytes"
+	"encoding/json"
+
+	"github.com/evergreen-ci/evergreen/rest/client"
+	system_metrics "github.com/evergreen-ci/timber/system_metrics"
+	"github.com/mongodb/ftdc/metrics"
+	"github.com/pkg/errors"
+	"github.com/shirou/gopsutil/cpu"
+	"github.com/shirou/gopsutil/disk"
+	"github.com/shirou/gopsutil/host"
+	"github.com/shirou/gopsutil/process"
+)
+
+// MetricCollector is an interface representing an object that can collect
+// a single system metric over a series of time steps.
+type MetricCollector interface {
+	// Name returns a string indicating the type of metric collected, such as "uptime".
+	Name() string
+	// Format returns the format of the collected data.
+	Format() DataFormat
+	// Collect returns the value of the collected metric when the function is called.
+	Collect() ([]byte, error)
+}
+
+type SystemMetricCollector struct {
+	mu          sync.Mutex
+	wg          sync.WaitGroup
+	innerCancel context.CancelFunc
+	logger      client.LoggerProducer
+	taskOpts    *system_metrics.SystemMetricsOptions
+	connOpts    system_metrics.ConnectionOptions
+	interval    time.Duration
+	collectors  []MetricCollector
+	id          string
+	client      *system_metrics.SystemMetricsClient
+	errored     bool
+	closed      bool
+}
+
+type DiskUsageCollector struct{}
```

Comment:
Is there a reason these three structs are public?

Reply:
done